flow360/component/simulation/primitives.py

```diff
@@ -122,7 +138,85 @@ class Box(_VolumeEntityBase):
     # pylint: disable=no-member
     center: LengthType.Point = pd.Field()
     size: LengthType.Point = pd.Field()
-    axes: Tuple[Axis, Axis] = pd.Field()
+    axis_of_rotation: Axis = pd.Field()
+    angle_of_rotation: AngleType = pd.Field()
+    private_attribute_input_cache: BoxCache = pd.Field(BoxCache(), frozen=True)
+
+    # pylint: disable=no-self-argument, missing-function-docstring
+    @MultiConstructorBaseModel.model_constructor
+    @pd.validate_call
+    def from_axes(
+        cls, name: str, center: LengthType.Point, size: LengthType.Point, axes: Tuple[Axis, Axis]
+    ):
+        # validate
+        x_axis, y_axis = np.array(axes[0]), np.array(axes[1])
+        x_axis /= np.linalg.norm(x_axis)
+        y_axis /= np.linalg.norm(y_axis)
+        z_axis = np.cross(x_axis, y_axis)
+        if not np.isclose(np.linalg.norm(z_axis), 1):
+            raise ValueError("Box axes not orthogonal.")
+
+        rotation_matrix = np.transpose(np.asarray([x_axis, y_axis, z_axis], dtype=float))
+
+        # Calculate the rotation axis n
+        eig_rotation = eig(rotation_matrix)
+        axis = np.real(eig_rotation[1][:, np.where(np.isclose(eig_rotation[0], 1))])
+        print(axis.shape)
+        if axis.shape[2] > 1:  # in case of 0 rotation angle
+            axis = axis[:, :, 0]
+        axis = np.ndarray.flatten(axis)
+
+        angle = np.sum(abs(np.angle(eig_rotation[0]))) / 2
+
+        # Find correct angle
+        matrix_test = cls._rotation_matrix_from_axis_and_angle(axis, angle)
+        angle *= -1 if np.isclose(rotation_matrix[0, :] @ matrix_test[:, 0], 1) else 1
+
+        # pylint: disable=not-callable
+        return cls(
+            name=name,
+            center=center,
+            size=size,
+            axis_of_rotation=list(axis),
+            angle_of_rotation=angle * u.rad,
+        )
+
+    # pylint: disable=no-self-argument
+    @classmethod
+    def _rotation_matrix_from_axis_and_angle(cls, axis, angle):
```

Comment:
Can we move this out of box class as a standalone function?

Reply:
Where should this function be?